cmd/restic/global.go

```diff
@@ -493,10 +493,7 @@ func OpenRepository(ctx context.Context, opts GlobalOptions) (*repository.Reposi
 		}
 	}
 	if err != nil {
-		if errors.IsFatal(err) {
-			return nil, err
-		}
-		return nil, errors.Fatalf("%s", err)
+		return nil, err
```

Comment:
This still changes the error handling when loading the repository key does not work. I'd like to keep the current fatal error handling for those cases.

Either change this to
```suggestion
		if errors.IsFatal(err) || errors.Is(err, repository.ErrNoKeyFound) {
		    return nil, err
		}
		return nil, errors.Fatalf("%s", err)
```
or
```suggestion
		if errors.IsFatal(err) {
		    return nil, err
		}
		return nil, fmt.Errorf("Fatal: %w", err)
```

Reply:
Ah gotcha - yeah, I can make that change. Suggestion 2 feels slightly cleaner to me.

But can I also step back and ask about the thinking around FatalError? I kind of like it as a concept, and still like my original fix here that added wrapping support.

FatalError solves one problem currently, and could be augmented to do more:
1. (currently) It flags errors as "expected" so that when `main.go` gets an exit error that isn't Fatal, we know to add extra debugging info.
2. (future) It could hold an exit code (from a centrally maintained list), so that `main.go` doesn't need to have its own switch/case list of them. This sort of thing would be particularly handy in the future, when a EACCES syscall errno in one part of the code might mean exit code 72, but in another part of the code would be exit code 76.  You'd have to wrap the the syscall.Errno in a custom error anyway so that `main.go` could disambiguate the cases.

If you are mentally deprecating FatalError, I imagine that `main.go`'s switch/case list is going to keep growing for both those use cases. I personally vibe with the FatalError approach over that.

Reply:
OK, this PR is now up-to-date and includes suggestion 1.

Happy to land this while also nattering about FatalError.

Reply:
Hmm, another advantage of FatalError is that it’s easier to test, versus main.go

Reply:
> If you are mentally deprecating FatalError, I imagine that `main.go`'s switch/case list is going to keep growing for both those use cases. I personally vibe with the FatalError approach over that.

I haven't though particularly much about how a replacement for the current FatalError should look like. The only thing that's clear is that the current "convert everything into a string"-approach doesn't work. But I also agree that the switch/case list in main.go is already a mess.

> 2\. It could hold an exit code (from a centrally maintained list), so that `main.go` doesn't need to have its own switch/case list of them.

Sounds reasonable. I'm not entirely sure yet which layer should make that translation. Should it already happen in `internal/` or wait until the `cmd/restic/` code? Just blindly wrapping all errors also leaks quite a lot of implementation details, which probably should be avoided.

Quite a lot of code currently just passes an error up the call chain without wrapping it in any way, which makes it rather hard to pinpoint the origin. So, we'll need some mix of both. Not sure how that will look in the end.

The main reason why I don't want these changes in this PR is that they are too invasive for inclusion in a patch release.